src/Symfony/Bundle/SecurityBundle/Security/FirewallMap.php

```diff
@@ -28,7 +28,7 @@ class FirewallMap implements FirewallMapInterface
     protected $map;
     private $contexts;
 
-    public function __construct(ContainerInterface $container, array $map)
+    public function __construct(ContainerInterface $container, $map)
```

Comment:
I doubt many users are actually overriding this class (perhaps JMS security bundle?) but I would do it only in 4.0 with proper docs.